File: src/main/java/com/myplaylists/dto/YoutubeDto.java
package com.myplaylists.dto;

import lombok.Builder;
import lombok.Getter;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

@Getter
public class YoutubeDto {

    private List<Map<String, Object>> songs;

    public YoutubeDto() {
        songs = new ArrayList<>();
    }

    public void add(Map<String, Object> song) {
        songs.add(song);
    }
}

File: src/main/kotlin/com/myplaylists/dto/YoutubeDto.kt
package com.myplaylists.dto

class YoutubeDto(
    var songs: ArrayList<Map<String, *>> = arrayListOf()
): BaseResponse() {
    fun add(song: Map<String, *>) = songs.add(song)
}

class YoutubeResponse(
    val items: List<YoutubeItem>
) {
}

class YoutubeItem(
    val id: ItemId,
    val snippet: Snippet,
)

class ItemId(
    val videoId: String,
)

class Snippet(
    val title: String
)